network/p2p/libp2pNode.go

```diff
@@ -109,3 +115,38 @@ type PeerConnections interface {
 	// 	  to the peer is not empty. This indicates a bug within libp2p.
 	IsConnected(peerID peer.ID) (bool, error)
 }
+
+// DisallowListNotificationConsumer is an interface for consuming disallow/allow list update notifications.
+type DisallowListNotificationConsumer interface {
+	// OnDisallowListNotification is called when a new disallow list update notification is distributed.
+	// Any error on consuming event must handle internally.
+	// The implementation must be concurrency safe.
+	// Args:
+	// 	id: peer ID of the peer being disallow-listed.
+	// 	cause: cause of the peer being disallow-listed (only this cause is added to the peer's disallow-listed causes).
+	// Returns:
+	// 	none
+	OnDisallowListNotification(id peer.ID, cause network.DisallowListedCause)
+
+	// OnAllowListNotification is called when a new allow list update notification is distributed.
+	// Any error on consuming event must handle internally.
+	// The implementation must be concurrency safe.
+	// Args:
+	// 	id: peer ID of the peer being allow-listed.
+	// 	cause: cause of the peer being allow-listed (only this cause is removed from the peer's disallow-listed causes).
+	// Returns:
+	// 	none
+	OnAllowListNotification(id peer.ID, cause network.DisallowListedCause)
+}
+
+// DisallowListOracle is an interface for querying disallow-listed peers.
+type DisallowListOracle interface {
+	// GetAllDisallowListedCauses for a disallow-listed peer returns all disallow-listed causes.
+	// If the peer is not disallow-listed, returns an empty slice (not nil).
+	// The implementation must be concurrency safe.
+	// Args:
+	// 	none
+	// Returns:
+	// 	[]network.DisallowListedCause: list of disallow-listed causes for the peer or empty slice if the peer is not disallow-listed.
+	GetAllDisallowListedCauses(peerId peer.ID) []network.DisallowListedCause
```

Comment:
```suggestion
	// IsDisallowListed for a disallow-listed peer returns true and all disallow-listed causes.
	// If the peer is not disallow-listed, returns false and nil.
	// The implementation must be concurrency safe.
	// Args:
	// 	none
	// Returns:
	// 	[]network.DisallowListedCause: list of disallow-listed causes for the peer or empty slice if the peer is not disallow-listed.
	// 	bool: true if the peer is disallow-listed false otherwise.
	IsDisallowListed(peerId peer.ID) ([]network.DisallowListedCause, bool)
```

It looks like we are using this function to check if the peer is disallow-listed, we should reflect that more clearly in the naming and more intuitive function usage.